infra/discovery/src/apollo/app.js

```diff
@@ -36,10 +36,14 @@ app.use(bundle)
 const server = new ApolloServer({
   resolvers,
   typeDefs,
-  context: async () => {
+  context: async context => {
+    const headers = context.req.headers
+    if (headers['x-discovery-auth-token']) {
```

Comment:
Please add this header to the [ingress config for discovery.](https://github.com/OriginProtocol/origin/blob/master/devops/kubernetes/charts/origin/templates/discovery.ingress.yaml) Also the cors configuration so that admin.originprotocol.com can query discovery.originprotocol.com.

Here is an example: https://github.com/OriginProtocol/origin/blob/master/devops/kubernetes/charts/origin/templates/growth.ingress.yaml

This could be done as a separate PR.